service.go

```diff
@@ -143,3 +143,63 @@ func (s *Service) SearchCertsForComponent(clusterID, componentName string) (Asse
 		}
 	}
 }
+
+// Watches secrets in namespace. If namespace is empty string this function will look up secrets in all namespaces.
+func (s *Service) SearchCertsForComponentInNamespace(namespace, clusterID, componentName string) (AssetsBundle, error) {
+	// TODO we should also do a list. In case the secrets have already been
+	// created we might miss them with only watching.
+	watcher, err := s.k8sClient.Core().Secrets(namespace).Watch(apismetav1.ListOptions{
```

Comment:
What do you think about having a private method that is used by both public methods to not have this code duplicated so much? It is way too easy to fix bugs in one place and forget the other.

Reply:
I also dont like this. Will create.